### ISSUE TITLE
[Arch+DDE] Suspend is not working

### ISSUE THREAD

**Issue by haonanq666** [linuxdeepin/developer-center#1998]
Perhaps related to #1980 as deepin v20 changes were pushed over to dde for arch. 

After the latest update (```pacman -Syu``` was run hours before posting), closing the lid of laptop no longer suspends the computer. ```journalctl -f``` gives the output

```
daemon/session/power[812]: utils.go:138: can not suspend
```

Also, the power icon in the tray and 启动器 (launcher? idk its name in English) does nothing when clicked. 

Running ```systemctl suspend``` does successfully suspend the laptop, but dde wakes from suspend after a glitchy sequence (screen flashes once, kwin compositor restarting due to graphics reset (this issue existed before too)) and does not ask for password to login. In my settings, this option is enabled. 

![image](https://user-images.githubusercontent.com/52809502/101568067-f781e300-399f-11eb-8962-c08c4d6350a1.png)
After setting the settings as such, I waited a minute. Screen turned off as a minute passed, but computer did not suspend. Pressing any key turned the screen back on immediately, and did not ask for password even though it should have. 
![image](https://user-images.githubusercontent.com/52809502/101568208-344dda00-39a0-11eb-82a3-f425f8050495.png)

Perhaps related to the issue, I noticed that despite setting a background for the login screen, the login screen displays the stock background. 

(writing this bug report, I noticed ctrl+alt+A shortcut for dde-screen-recorder doesn't work again.)

**Comment by haonanq666** [comment]
更新：经过反复测试，发现如下方法修复：

1: 先将```startdde``` 退回至```5.6.0.10``` （需要退两个版本，上一版有登录后闪退的问题）。降级后关盖可以待机，但是桌面渲染有卡顿、乱码等问题。

2: 将```deepin-desktop-schemas``` 退回至```5.8.0.21``` （退两版原因同上）。降级后渲染也没问题了，可近似正常使用，但是从待机后唤醒仍不要求输入密码。

3: 将```deepin-session-shell```退回至```5.3.0.32``` 唤醒后会要求输入密码，系统基本正常。

4: 发现有时候从待机唤醒时```kwinrc``` 里面内容改成了```OpenGLisUnsafe=true```和```Enabled=false```。将```kwin```降级至```5.20.3```应该好了，目前compositor能用，但是未仔细测试。

```kwin``` 问题是老问题了，应该是我瞎升级降级版本加剧了问题，所以降级后更匹配一点？再具体的技术问题我就不知道了。。。

**Comment by haonanq666** [comment]
Latest update does not fix; suspend not working with latest startdde, and cannot lock screen on suspend with latest deepin-session-shell. 

In deepin-session-shelll version 5.3.0.32, switching to tty causes the graphical session to be logged out; this causes the tty to automatically switch back to graphical session. (i.e. am logged into graphical session, press ctrl+alt+F2,  and reach tty2, but two seconds later screen automatically switches back to tty1 which now shows the lock screen). Also, on this version of packages, compositing breaks every time I lock screen. Current fix is to kill x11 every time I log in. 

Please prioritize fixing startdde and deepin-session-shell as this is grossly detracting from user experience at the moment.

**Comment by KimHono** [comment]
Hi !
I'm not using auto suspend after a predefined delay, but I'm using this command everyday manually.
Luckily, the session does suspend and wake up without failure. Moreover, a login password is always required.

Here are my packages:
`deepin-session-shell` 5.3.0.39-1
`startdde` 5.6.0.34-1
`deepin-desktop-schemas` 5.8.0.42-1
`kwin` 5.20.4-1
`ARCH Linux` 5.10.3-arch1-1

**Comment by haonanq666** [comment]
yes, session-shell got fixed, but startdde is still kinda broken. ```systemctl suspend``` does indeed work, but lid closing or auto suspend does not.

**Comment by franciscocpg** [comment]
I'm having the same issue. When looking at `journalctl` I see the logs above. So it looks like is something related to `com.deepin.daemon.PowerManager`?
```
jan 14 12:49:16 francisco-jusbrasil daemon/session/power[1311]: utils.go:133: The name com.deepin.daemon.PowerManager was not provided by any .service files
jan 14 12:49:16 francisco-jusbrasil daemon/session/power[1311]: lid_switch.go:64: Lid closed
jan 14 12:49:16 francisco-jusbrasil daemon/system/power[836]: lid_switch.go:37: Lid Closed
jan 14 12:49:16 francisco-jusbrasil dde-system-daemon[836]: <info> lid_switch.go:37: Lid Closed
jan 14 12:49:16 francisco-jusbrasil systemd-logind[713]: Lid closed.
```
